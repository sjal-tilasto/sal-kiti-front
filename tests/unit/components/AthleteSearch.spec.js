import mockAxios from "axios";
import { createLocalVue, shallowMount, mount } from "@vue/test-utils";
import SearchPage from "@/components/AthleteSearch.vue";
import BootstrapVue from "bootstrap-vue";
import flushPromises from "flush-promises";
import Vuex from "vuex";

const localVue = createLocalVue();
localVue.use(BootstrapVue);

const $router = {
  push: jest.fn()
};

const state = { user: { is_authenticated: false } };
const store = new Vuex.Store({ state });

describe("AthleteSearch.vue", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("renders search form", () => {
    const $route = {
      query: {}
    };

    const wrapper = shallowMount(SearchPage, {
      localVue,
      store,
      mocks: {
        $route
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("parses query parameter", () => {
    const $route = {
      query: {
        search: "athlete"
      }
    };
    const wrapper = shallowMount(SearchPage, {
      localVue,
      store,
      mocks: {
        $route
      }
    });
    expect(wrapper.vm.form.search).toBe("athlete");
  });

  it("search triggers api call ", async () => {
    const event = { preventDefault: jest.fn() };
    const $route = {
      query: {
        search: "athlete"
      }
    };

    const wrapper = mount(SearchPage, {
      localVue,
      store,
      mocks: {
        $route,
        $router
      }
    });
    const input = wrapper.find("input");
    input.element.value = "athlete";
    wrapper.vm.onSubmit(event);
    await flushPromises();
    expect(mockAxios.get).toHaveBeenCalledWith(
      "athletes/?search=athlete&limit=10&page=1"
    );
    await flushPromises();
    expect(wrapper.vm.results.results[0].last_name).toBe("Archer");
  });

  it("render error message on error", async () => {
    mockAxios.get.mockImplementationOnce(() =>
      Promise.reject({
        response: {
          status: 404,
          statusText: "Not Found"
        }
      })
    );
    const $route = {
      query: {
        search: "Athlete",
        load: true
      }
    };
    const wrapper = shallowMount(SearchPage, {
      localVue,
      store,
      mocks: {
        $route,
        $router
      }
    });
    await flushPromises();
    expect(wrapper.text()).toContain("import.error.notfound");
  });
});
