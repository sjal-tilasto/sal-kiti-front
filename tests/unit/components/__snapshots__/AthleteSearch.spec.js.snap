// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AthleteSearch.vue renders search form 1`] = `
<div>
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <h1>
        athlete.athlete
      </h1>
    </b-col-stub>
  </b-row-stub>
   
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <!---->
    </b-col-stub>
  </b-row-stub>
   
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <h2
        class="bg-sal-orange"
      >
        search.athlete
      </h2>
       
      <b-form-stub
        inline="true"
      >
        <b-form-input-stub
          aria-label="search.placeholder_athlete"
          class="search-field space-right"
          debounce="0"
          formatter="function bvConfigurablePropDefault() {
      var value = getComponentConfig(componentKey, key, prop.default);
      return isFunction(value) ? value() : value;
    }"
          placeholder="search.placeholder_athlete"
          type="text"
          value=""
        />
         
        <b-form-select-stub
          aria-label="search.max_results"
          class="search-field space-right"
          disabledfield="disabled"
          htmlfield="html"
          labelfield="label"
          options=""
          optionsfield="options"
          selectsize="0"
          textfield="text"
          value="10"
          valuefield="value"
        >
          <option
            value="10"
          >
            
            10
          
          </option>
          <option
            value="25"
          >
            
            25
          
          </option>
          <option
            value="50"
          >
            
            50
          
          </option>
          <option
            value="100"
          >
            
            100
          
          </option>
        </b-form-select-stub>
         
        <b-button-stub
          class="btn-orange space-right"
          tag="button"
          target="_self"
          type="submit"
          variant="light"
        >
          search.search
        </b-button-stub>
      </b-form-stub>
    </b-col-stub>
  </b-row-stub>
   
  <br />
   
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <h3
        class="bg-sal-orange"
      >
        result.result
      </h3>
       
      <!---->
       
      <!---->
       
      <b-table-stub
        currentpage="1"
        emptyfilteredtext="There are no records matching your request"
        emptytext="There are no records to show"
        fields="[object Object],[object Object],[object Object],[object Object]"
        filterdebounce="0"
        filterfunction="function bvConfigurablePropDefault() {
      var value = getComponentConfig(componentKey, key, prop.default);
      return isFunction(value) ? value() : value;
    }"
        filterignoredfields=""
        filterincludedfields=""
        hover="true"
        id="my-table"
        items=""
        labelsortasc="Click to sort Ascending"
        labelsortclear="Click to clear sorting"
        labelsortdesc="Click to sort Descending"
        perpage="0"
        selectedvariant="active"
        selectmode="multi"
        sortcompare="function bvConfigurablePropDefault() {
      var value = getComponentConfig(componentKey, key, prop.default);
      return isFunction(value) ? value() : value;
    }"
        sortcompareoptions="[object Object]"
        sortdirection="asc"
        value=""
      />
       
      <div
        style="display: none;"
      >
        <b-spinner-stub
          label="Loading..."
          role="status"
          tag="span"
          type="border"
        />
      </div>
    </b-col-stub>
  </b-row-stub>
</div>
`;
